en/includes/references/grant-types.md

```diff
@@ -181,42 +443,81 @@ The following diagram shows how the token exchange grant flow works.
-5. The Authorization Server validates the SAML 2.0 assertion and if valid, generates a new access token of the requested type based on the provided parameters and the server's policy, and responds to the client with the new access token.
+4. The client application makes a token exchange request to the authorization server.
+
+    === "Request format (/token)"
+
+        ```bash
+        curl -v -k -X POST {{base_url}}/oauth2/token \
+        --header "Authorization: Basic <Base64Encoded(CLIENT_ID:CLIENT_SECRET)>" \
+        --header "Content-Type:application/x-www-form-urlencoded" \
+        --data-urlencode "grant_type=urn:ietf:params:oauth:grant-type:saml2-bearer" \
+        --data-urlencode "assertion=<base64-URL_encoded_assertion>" \
+        --data-urlencode "scope=<scopes>" \
+        ```
+    
+    === "Sample request (/token)"
+
+        ```bash
+        curl -v -k -X POST {{base_url_example}}/oauth2/token \
+        --header "Authorization: Basic RWkwV2Y5YnpmTXE0UTBsZndTdlRQamU4a2NFYTpIRvvyUzJIUjlrZE9YMjBXTG9JNmY1eE1wdUlBamdKeG5aUVVUMV9lNTJnYQ==" \
+        --header "Content-Type:application/x-www-form-urlencoded" \
+        --data-urlencode "grant_type=urn:ietf:params:oauth:grant-type:saml2-bearer" \
+        --data-urlencode "assertion=PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZczpUcmFuc2Zvcm1zPgo8ZHM6VHJhbnNmb3JtIEFsZ29yaXRobT0iaHR0cDovL3d3zOlRyYW5zZm9ybXMaW" \
+        --data-urlencode "scope=internal_idp_view internal_idp_create internal_organization_view"
+        ```
+
+5. The Authorization Server validates the SAML 2.0 assertion and if valid, generates a new access token and responds to the client with the new access token.
 6. The client application can now request resources from the resource server by providing the access token.
 7. As the resource server trusts {{ product_name }} issued tokens, it returns the requested resources to the client application.
+{% endif %}
 
 ## Organization switch grant
 
-The organization switch grant is a custom grant type in {{ product_name }} that enables users to switch between different organizations in a hierarchical organization structure.
-
-Client applications should always use one of the traditional grant types to authorize user access. The organization switch grant is also required when the authorization request is for resources of an organization. That is because it is necessary to switch between the organization (root) and the organization to obtain access tokens that are valid for organizations.
+The organization switch grant is a custom grant type in {{ product_name }} that enables users to switch between [organizations]({{base_path}}/guides/organization-management/) that exists in a hierarchical structure. The process involves using the authorization code grant to obtain an access token for the root organization and exchanging it to an access token usable within a child organization.
```

Comment:
```suggestion
The organization switch grant is a custom grant type in {{ product_name }} that enables users to switch between [organizations]({{base_path}}/guides/organization-management/) that exists in a hierarchical structure.
```

Reply:
Shall we remove this because it id not the only path which works for organization switch

Reply:
we can add this info below, because digram illustrates that case